src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheck.java

```diff
@@ -248,15 +249,15 @@ private void collectReferencesFromJavadoc(DetailAST ast) {
      */
     private static Set<String> collectReferencesFromJavadoc(TextBlock textBlock) {
         final Set<String> references = new HashSet<>();
-        // process all the @link type tags
+        final List<JavadocTag> tags = new ArrayList<>();
+        // gather all the inline tags, like @link
         // INLINE tags inside BLOCKs get hidden when using ALL
-        getValidTags(textBlock, JavadocUtils.JavadocTagType.INLINE).stream()
+        tags.addAll(getValidTags(textBlock, JavadocUtils.JavadocTagType.INLINE));
+        // gather all the block-level tags, like @throws and @see
+        tags.addAll(getValidTags(textBlock, JavadocUtils.JavadocTagType.BLOCK));
+        tags.stream()
             .filter(JavadocTag::canReferenceImports)
             .forEach(tag -> references.addAll(processJavadocTag(tag)));
-        // process all the @throws type tags
-        getValidTags(textBlock, JavadocUtils.JavadocTagType.BLOCK).stream()
-            .filter(JavadocTag::canReferenceImports)
-            .forEach(tag -> references.addAll(matchPattern(tag.getFirstArg(), FIRST_CLASS_NAME)));
```

Comment:
The real change is here: replacing `matchPattern(..)` with the same `processJavadocTag(tag)` that INLINE uses. 

After that change, both INLINE and BLOCK used the same `.filter` and `.foreach`, so the rest of the diff is to remove the duplication.